content/800-data-platform/100-accelerate/300-concepts.mdx

```diff
@@ -0,0 +1,192 @@
+
+Use `ttl` in `cacheStrategy` and specify the TTL of the query in seconds.
+
+```javascript
+await prisma.user.findMany({
+  cacheStrategy: {
++    ttl: 60,
+  },
+});
+```
+
+With a specified TTL of 60 seconds, the majority of requests will result in
+a cache hit throughout the TTL duration.
+
+![TTL](./images/ttl.png)
+
+TTL is useful for reducing database load and latency for data that does not require frequent updates.
+
+## Stale-While-Revalidate (SWR)
+
+Stale-While-Revalidate (SWR) allows you to control how long Accelerate can serve stale cache data while fetching fresh data in the background. When a read query is executed, Accelerate checks the age of the cached response against the `swr` duration. If the cache data is within the `swr` limit, Accelerate serves the stale data while simultaneously refreshing the cache by fetching the latest data from the database.
+
+Use `swr` in `cacheStrategy` and specify the SWR of the query in seconds.
+
+```javascript
+await prisma.user.findMany({
+  cacheStrategy: {
++    swr: 60,
+  },
+});
+```
+
+When specifying a SWR of 60 seconds, the cache serves stale data until the
+cache refreshes itself in the background after each request.
+
+![SWR](./images/swr.png)
+
+After the cache is refreshed in the background, the `swr` timer resets and the data is once again available in the cache for the duration of the `swr` period. As a result, most results will be fetched from the cache because the `swr` duration resets after each request, and a roundtrip to the database is avoided.
+
+## Time-to-live and Stale-While-Revalidate combination
+
+You can use both TTL and SWR and stack them for the same query. SWR applies _after_ TTL has expired.
+
+Accelerate returns previously cached data up to the age specified in TTL without revalidating. After TTL expires, Accelerate serves the cached data as stale up to the age specified in SWR and revalidates it in the background.
+
+It is useful to combine TTL and SWR to maintain low latency while Accelerate refreshes cached data in the background.
+
+Use `ttl` and `swr` in `cacheStrategy` and specify the TTL and SWR of the query in seconds.
+
+```javascript
+await prisma.user.findMany({
+  cacheStrategy: {
++    ttl: 60,
++    swr: 30,
+  },
+});
+```
+
+When specifying a TTL of 30 seconds and SWR of 60 seconds, the cache serves
+fresh data for the initial 30 seconds. Subsequently, it serves stale data
+until the cache refreshes itself in the background after each request.
+
+![TTL and SWR](./images/ttl_and_swr.png)
```

Comment:
```suggestion
With a specified TTL of 60 seconds, the majority of requests will result in
a cache hit throughout the TTL duration:

![TTL](./images/ttl.png)

TTL is useful for reducing database load and latency for data that does not require frequent updates.

## Stale-While-Revalidate (SWR)

Stale-While-Revalidate (SWR) allows you to control how long Accelerate can serve stale cache data while fetching fresh data in the background. When a read query is executed, Accelerate checks the age of the cached response against the `swr` duration. If the cache data is within the `swr` limit, Accelerate serves the stale data while simultaneously refreshing the cache by fetching the latest data from the database.

Use `swr` in `cacheStrategy` and specify the SWR of the query in seconds:

```javascript
await prisma.user.findMany({
  cacheStrategy: {
+    swr: 60,
  },
});
```

When specifying a SWR of 60 seconds, the cache serves stale data until the
cache refreshes itself in the background after each request:

![SWR](./images/swr.png)

After the cache is refreshed in the background, the `swr` timer resets and the data is once again available in the cache for the duration of the `swr` period. As a result, most results will be fetched from the cache because the `swr` duration resets after each request, and a roundtrip to the database is avoided.

## Time-to-live and Stale-While-Revalidate combination

You can use both TTL and SWR and stack them for the same query. SWR applies _after_ TTL has expired.

Accelerate returns previously cached data up to the age specified in TTL without revalidating. After TTL expires, Accelerate serves the cached data as stale up to the age specified in SWR and revalidates it in the background.

It is useful to combine TTL and SWR to maintain low latency while Accelerate refreshes cached data in the background.

Use `ttl` and `swr` in `cacheStrategy` and specify the TTL and SWR of the query in seconds:

```javascript
await prisma.user.findMany({
  cacheStrategy: {
+    ttl: 60,
+    swr: 30,
  },
});
```

When specifying a TTL of 30 seconds and SWR of 60 seconds, the cache serves
fresh data for the initial 30 seconds. Subsequently, it serves stale data
until the cache refreshes itself in the background after each request:

![TTL and SWR](./images/ttl_and_swr.png)
```